### DISCUSSION TITLE
How can we speed up FindMarkers

### DISCUSSION THREAD

**Discussion by KANG-BIOINFO** [satijalab/seurat#4433] [Q&A] [answered]
Hi,

Thanks for the awesome package for single-cell analysis. I was using FindAllMarkers function and found the marker identification is slower than the corresponding function of Scanpy. Usually for a data with tens of thousands cells (e.g. 50K cells), this function would take more than 10 minutes to finish. I tried to use _future_ for parallel computation, but the improvement is not very big. May I ask if there're plans for improvement of the running speed for FindAllMarkers?

Thanks!

**Comment by timoast** [comment, marked as answer]
You can try using [presto](https://github.com/immunogenomics/presto) which is a much faster implementation of wilcoxon DE testing, and is also available through [seurat-wrappers](https://github.com/satijalab/seurat-wrappers)

**Reply by alextamburino** [reply]
@timoast Has anyone noticed a performance change when moving to Seurat v4? I just moved ~1 month ago and FindAllMarkers has been taking a lot longer with the same cell numbers and memory.

**Reply by timoast** [reply]
I haven't noticed this...

**Reply by tzvi909** [reply]
How about for the ROC method though?
